ruzzle_bare_minimum.py

```diff
@@ -79,32 +79,36 @@ def dfs(self, visited, s, word, word_pts, word_mult, path):
         len_word = len(word)
 
         # store all >2 letter possible words in words_info if they are actual words
-        if len_word >= PREFIX_LOWER_BOUND:
-            # ex) if a word begins with TTH, this doesn't exist so we can stop adding letters
-            if len_word <= PREFIX_UPPER_BOUND and word not in PREFIXES[len_word - PREFIX_LOWER_BOUND]:
-                return
-
-            if word in DICTIONARY:
-                score = word_pts * word_mult
-                bonus = 0 if len_word < 4 else 5 * (len_word - 4)  # length bonus
-                self.possible_words.append((word, score + bonus, path[:]))  # append copy of path
-
-            # there are no words greater than 12 letters (based on ruzzle database), so stop searching
-            if len_word == MAX_WORD_LEN:
-                return
+        if len_word >= PREFIX_LOWER_BOUND and word in DICTIONARY:
```

Comment:
I thought PREFIX_LOWER_BOUND was for deciding which prefixes to check between 2 and 8. Here, doesn't PREFIX_LOWER_BOUND set the minimum length of word that is found?

Reply:
Good point - I decided to add another variable for the minimum word length.